gateway-service/src/main/java/org/zowe/apiml/gateway/filters/pre/OIDCAuthenticationFilter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.filters.pre;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.AuthenticationFailureHandler;
+import org.zowe.apiml.gateway.security.service.schema.source.AuthSource;
+import org.zowe.apiml.gateway.security.service.schema.source.OIDCAuthSource;
+import org.zowe.apiml.gateway.security.service.schema.source.OIDCAuthSourceService;
+import org.zowe.apiml.security.common.login.NonCompulsoryAuthenticationProcessingFilter;
+import org.zowe.apiml.security.common.token.TokenAuthentication;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Optional;
+
+@Slf4j
+public class OIDCAuthenticationFilter extends NonCompulsoryAuthenticationProcessingFilter {
+
+    private final AuthenticationFailureHandler failureHandler;
+
+    private final OIDCAuthSourceService oidcAuthSourceService;
+
+    public OIDCAuthenticationFilter(String endpoint,
+                                    AuthenticationFailureHandler failureHandler,
+                                    OIDCAuthSourceService oidcAuthSourceService) {
+        super(endpoint);
+        this.failureHandler = failureHandler;
+        this.oidcAuthSourceService = oidcAuthSourceService;
+
+
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) {
+        Optional<String> accessToken = oidcAuthSourceService.getToken(request);
+        if (accessToken.isPresent()) {
+            log.debug("OIDC access token found in the request");
+            AuthSource authSource = new OIDCAuthSource(accessToken.get());
+            AuthSource.Parsed parsedToken = oidcAuthSourceService.parse(authSource);
+            if (parsedToken != null && StringUtils.isNotEmpty(parsedToken.getUserId())) {
+                return TokenAuthentication.createAuthenticated(parsedToken.getUserId(), accessToken.get());
+            }
+        }
+
+        log.debug("No valid OIDC access token found in the request.");
```

Comment:
Should it rather be error?

Reply:
It's just a debug message saying that this filter was applied but no access token was found in the request. If it is error message then the log would be flooded with this message everytime when request is not made with OIDC access token in it.

Reply:
Ok I see, I'm ok with debug then